src/com/google/enterprise/adaptor/opentext/OpentextAdaptor.java

```diff
@@ -1032,8 +1055,23 @@ void doDocument(DocumentManagement documentManagement,
           "Document does not support versions: " + opentextDocId);
     }
 
-    Version version = documentManagement.getVersion(documentNode.getID(),
+    Version version;
+    try {
+      version = documentManagement.getVersion(documentNode.getID(),
         this.currentVersionType);
+    } catch (SOAPFaultException soapFaultException) {
+      if (isPermissionsFailure(soapFaultException)) {
+        log.log(Level.FINE,
+            soapFaultException.getFault().getFaultString() + " {0}",
+            opentextDocId.toString());
+        response.respondNotFound();
+      } else {
+        log.log(Level.WARNING,
+            "Error retrieving version: " + opentextDocId,
+            soapFaultException);
+      }
+      return;
```

Comment:
So this used to throw a non-permissions exception, was it all the way out (i.e., a 50x)? Now what happens with this return?